src/librustc_typeck/outlives/mod.rs

```diff
@@ -96,27 +97,43 @@ fn inferred_outlives_crate(
     let predicates = global_inferred_outlives
         .iter()
         .map(|(&def_id, set)| {
-            let predicates = tcx.arena.alloc_from_iter(set
+            let def_span = tcx.def_span(def_id);
+            let generics = tcx.generics_of(def_id);
+            let predicates = &*tcx.arena.alloc_from_iter(set
                 .iter()
                 .filter_map(
                     |ty::OutlivesPredicate(kind1, region2)| match kind1.unpack() {
                         GenericArgKind::Type(ty1) => {
-                            Some(ty::Predicate::TypeOutlives(ty::Binder::bind(
+                            // FIXME(eddyb) compute `Span`s in `implicit_infer`.
```

Comment:
is this FIXME still relevant?

Reply:
Yeah, because the inference uses a set, I was worried I would introduce many copies of the same predicate and cause perf issues for everything looking at the (larger) list of predicates.

Doing this without perf regressions would be trickier, probably...
<hr>

Oh wait I can replace `BTreeSet<Predicate>` with `BTreeMap<Predicate, Span>` and just keep the first `Span` any one `Predicate` is found at.
Let me do that once the first perf results are in (since we'd need another perf run afterwards).